tests/relay/test_rln.py

```diff
@@ -1,24 +1,132 @@
-import os
+import math
+import random
+from time import time
 import pytest
-
-from src.env_vars import RLN_CREDENTIALS
+from src.libs.common import delay, to_base64
 from src.libs.custom_logger import get_custom_logger
-from src.steps.relay import StepsRelay
+from src.steps.rln import StepsRLN
+from src.test_data import SAMPLE_INPUTS
 
 logger = get_custom_logger(__name__)
 
 
-@pytest.mark.usefixtures()
-class TestRelayRLN(StepsRelay):
-    def test_register_rln(self):
-        logger.debug("Running register RLN test for main relay nodes")
-        key_stores_found = 0
+@pytest.mark.usefixtures("register_main_rln_relay_nodes")
+class TestRelayRLN(StepsRLN):
+    def test_publish_with_valid_payloads_at_slow_rate(self):
+        self.setup_main_rln_relay_nodes()
+        self.subscribe_main_relay_nodes()
+        failed_payloads = []
+        for payload in SAMPLE_INPUTS:
+            logger.debug(f'Running test with payload {payload["description"]}')
+            message = self.create_message(payload=to_base64(payload["value"]))
+            try:
+                self.check_published_message_reaches_relay_peer(message)
+            except Exception as e:
+                logger.error(f'Payload {payload["description"]} failed: {str(e)}')
+                failed_payloads.append(payload["description"])
+            delay(1)
+            assert not failed_payloads, f"Payloads failed: {failed_payloads}"
+
+    def test_publish_with_valid_payloads_at_spam_rate(self):
+        self.setup_main_rln_relay_nodes()
+        self.subscribe_main_relay_nodes()
+        previous = math.trunc(time())
+        for i, payload in enumerate(SAMPLE_INPUTS[:4]):
+            logger.debug(f'Running test with payload {payload["description"]}')
+            message = self.create_message(payload=to_base64(payload["value"]))
+            try:
+                now = math.trunc(time())
+                self.publish_message(message)
```

Comment:
Yes, they are discarded with "Bad request: assert "RLN validation failed" in str(e)